community/kernel/src/main/java/org/neo4j/kernel/impl/storageengine/impl/recordstorage/RecordStorageEngine.java

```diff
@@ -283,10 +268,13 @@ public void createCommands(
     {
         if ( txState != null )
         {
-            RecordChangeSet recordChangeSet = new RecordChangeSet( loaders );
-            TransactionRecordState recordState = new TransactionRecordState( neoStores, integrityValidator,
-                    recordChangeSet, lastTransactionIdWhenStarted, locks,
-                    relationshipCreator, relationshipDeleter, propertyCreator, propertyDeleter );
+            // We can make this cast here because we expected that the storageStatement passed in here comes from
```

Comment:
I think this is a common pattern for explicitly allocating instances (which is a good thing) and pass in instead of having the method allocate and manage itself. We own this code completely so it's never a practical problem.

Reply:
do not think that casting with a long comment is a _common pattern_ or either practice of some good sort